feat: bring back maxDiffPixels in toMatchSnapshot

aslushnikov opened the pull request:
This patch reverts 2 commits that removed the feature from the method:
- "fix: explicitly ignore maxDiffPixels in toMatchSnapshot (#12570)"
  commit b8af8458d614dcc53ef324553d63522448e94d94.
- "chore: remove `maxDiffPixels` from toMatchSnapshot (#12539)"
  commit a3dff45974d53087bd6ebea5164c75d5eb357e28.

rwoll commented on tests/playwright-test/golden.spec.ts:
```diff
@@ -394,6 +396,190 @@ test('should compare binary', async ({ runInlineTest }) => {
+        });
+      });
+    `
+  })).exitCode).toBe(1);
```
What does the error message look like? Can you add an expect for it to make sure it remains friendly?

aslushnikov replied:
added a confidence test!

dgozman commented on tests/playwright-test/golden.spec.ts:
```diff
@@ -394,6 +396,190 @@ test('should compare binary', async ({ runInlineTest }) => {
   expect(result.exitCode).toBe(0);
 });
 
+test('should throw for invalid maxDiffPixels values', async ({ runInlineTest }) => {
```
I'd drop all tests except two smoke tests, one for each option, and leave all the tests for toHaveScreenshot.

aslushnikov committed:
feat: bring back maxDiffPixels in toMatchSnapshot

This patch reverts 2 commits that removed the feature from the method:
- "fix: explicitly ignore maxDiffPixels in toMatchSnapshot (#12570)"
  commit b8af8458d614dcc53ef324553d63522448e94d94.
- "chore: remove `maxDiffPixels` from toMatchSnapshot (#12539)"
  commit a3dff45974d53087bd6ebea5164c75d5eb357e28.

aslushnikov committed:
fix lint
